src/Native/Unix/System.Security.Cryptography.Native/pal_ssl.cpp

```diff
@@ -524,3 +524,23 @@ extern "C" int32_t CryptoNative_SslAddExtraChainCert(SSL* ssl, X509* x509)
 
     return 0;
 }
+
+extern "C" int32_t CryptoNative_SslSelectNextProto(unsigned char** out, unsigned char* outlen, const unsigned char* server, unsigned int server_len, const unsigned char* client, unsigned int client_len)
+{
+    return SSL_select_next_proto(out, outlen, server, server_len, client, client_len);
+}
+
+extern "C" void CryptoNative_SslCtxSetAplnSelectCb(SSL_CTX* ctx, SslCtxSetAplnCallback cb, void* arg)
+{
+    SSL_CTX_set_alpn_select_cb(ctx, cb, arg);
+}
+
+extern "C" int32_t CryptoNative_SslCtxSetAlpnProtos(SSL_CTX* ctx, const unsigned char* protos, unsigned protos_len)
+{
+    return SSL_CTX_set_alpn_protos(ctx, protos, protos_len);
+}
+
+extern "C" void CryptoNative_SslGet0AlpnSelected(SSL* ssl, const unsigned char** protocol, unsigned int* len)
```

Comment:
These functions all need to use the correct type identifiers in their signatures.

* `char*` is only used for `string`.  `byte[]` => `uint8_t*`.
* `int` (C#) => `int32_t`

So they might be

```C++
// If the void*'s here can't be void* then whatever type you think they are should be encoded.  I'm guessing uint8_t.
extern "C" int32_t CryptoNative_SslSelectNextProto(void** out, uint8_t* outlen, const uint8_t* server, uint32_t server_len, const void* client, uint32_t client_len);

extern "C" int32_t CryptoNative_SslCtxSetAlpnProtos(SSL_CTX* ctx, const uint8_t* protos, uint32_t protos_len);

extern "C" void CryptoNative_SslGet0AlpnSelected(SSL* ssl, const uint8_t** protocol, uint32_t* len);
```